src/bitmap/inherent.rs

```diff
@@ -578,6 +578,80 @@ impl RoaringBitmap {
 
         None
     }
+
+    /// Removes the specified number of elements from the top.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use roaring::RoaringBitmap;
+    ///
+    /// let mut rb = RoaringBitmap::from_iter([1, 5, 7, 9]);
+    /// rb.remove_first(2);
+    /// assert_eq!(rb, RoaringBitmap::from_iter([7, 9]));
+    /// rb = RoaringBitmap::from_iter([1, 3, 7, 9]);
+    /// rb.remove_first(2);
+    /// assert_eq!(rb, RoaringBitmap::from_iter([7, 9]));
+    pub fn remove_first(&mut self, mut n: usize) {
+        if n > self.len() as usize {
+            return;
+        }
+        // remove containers up to the front of the target
+        let position = self.containers.iter().position(|container| {
+            let container_len = container.len() as usize;
+            if container_len < n {
+                n -= container_len;
+                false
+            } else {
+                true
+            }
+        });
+        // It is checked at the beginning of the function, so it is usually never an Err
+        let position = position.expect("there are no containers to delete");
+        if position > 0 {
+            self.containers.rotate_left(position);
+            self.containers.truncate(position);
+        }
+        // remove data in containers if there are still targets for deletion
+        if n > 0 {
+            // container immediately before should have been deleted, so the target is 0 index
+            self.containers[0].remove_first(n);
+        }
+    }
+
+    /// Removes the specified number of elements from the tail.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use roaring::RoaringBitmap;
+    ///
+    /// let mut rb = RoaringBitmap::from_iter([1, 5, 7, 9]);
+    /// rb.remove_last(2);
+    /// assert_eq!(rb, RoaringBitmap::from_iter([1, 5]));
+    /// rb.remove_last(1);
+    /// assert_eq!(rb, RoaringBitmap::from_iter([1]));
+    pub fn remove_last(&mut self, mut n: usize) {
+        if n > self.len() as usize {
+            return;
+        }
```

Comment:
Shouldn't we empty the whole map if we need to delete the last 20 integers in a map of 19 integers?